include/cglm/util.h

```diff
@@ -157,18 +172,113 @@ glm_clamp_zo(float val) {
 }
 
 /*!
- * @brief linear interpolation between two number
+ * @brief linear interpolation between two numbers
  *
- * formula:  from + s * (to - from)
+ * formula:  from + t * (to - from)
  *
  * @param[in]   from from value
  * @param[in]   to   to value
- * @param[in]   t    interpolant (amount) clamped between 0 and 1
+ * @param[in]   t    interpolant (amount)
  */
 CGLM_INLINE
 float
 glm_lerp(float from, float to, float t) {
-  return from + glm_clamp_zo(t) * (to - from);
+  return from + t * (to - from);
```

Comment:
The `t` parameter is clamped in all lerp functions (single float, vec4, vec3...). This breaks existing api consistency. I'm not sure if we should break existing api. On the other hand this will improve speed a bit which is good.

Options:
1. We could provide unclamped version which will not break api with another name? Like `glm_lerp_uc` **uc** suffix stands for un-clamped? Or you may come with better name
2. We could remove clamp from vec3 and vec4 and quat... too

Also it seems GLSL's mix functions don't apply clamping 🤔

Reply:
Yes, the GLSL specs do not apply clamping so I think the glm functions in order to have maximum consistency with GLSL should not apply clamping either. For example I would adopt a scheme such as the following: glm_lerp() [no clamping/standard] and then glm_lerpc() [with clamping]. This way we achieve maximum flexibility/efficiency by leaving it up to the user to pick the appropriate function (clamped but slower or non-clamped but faster). Same goes to vec3, vec4 and quat. So I would go with option 2. Existing users will have to revise their code.

As far as GLSL's mix goes: I have the implementation of a glm_mix() function, however I found it too similar to glm_lerp() to justify adding it and I ended up leaving it out. If you want, it would take me 5 mins to add it to a new PR.

Reply:
Ok then, let's go with 2.

> I found it too similar to glm_lerp() 

Actually it must be same, just different name convention.

`glm_mix` is really cool name rather than `glm_lerp`, especially for colors. Maybe we can provide both (`mix` as wrapper of `lerp`), so GLSL/Renderman users would use **mix** and HLSL users would use **lerp**.

Reply:
ok, sounds good to me.

Reply:
As lerp goes, now we also have glm_lerpc() which is the clamped version of glm_lerp(). Should I make a clamped version of glm_mix() as well? So in the end we would have glm_mix() wrapping glm_lerp() [non-clamped] and then glm_mixc() wrapping glm_lerpc() [clamped].

Reply:
yes, but I think `glm_mixc()` must be inline like `glm_mix()` because both are wrappers.

Reply:
Ok, I updated the PR.